plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerEditsBookScriptEvent.java

```diff
@@ -51,76 +49,67 @@ public class PlayerEditsBookScriptEvent extends BukkitScriptEvent implements Lis
     public PlayerEditsBookScriptEvent() {
         registerCouldMatcher("player edits book");
         registerCouldMatcher("player signs book");
+        this.<PlayerEditsBookScriptEvent>registerTextDetermination("not_signing", (evt) -> {
+            evt.event.setSigning(false);
+        });
+        this.<PlayerEditsBookScriptEvent, ScriptTag>registerOptionalDetermination(null, ScriptTag.class, (evt, context, value) -> {
+            if (value.getContainer() instanceof BookScriptContainer script) {
+                ItemTag dBook = script.getBookFrom(evt.getScriptEntryData().getTagContext());

```

Comment:
Sorry, one more thing (otherwise LGTM) - `registerOptionalDetermination` gives you a `context` parameter, no need to get the `TagContext` here again.